### PR TITLE
Logging cleanups

### PR DESCRIPTION
**Pull request by tongzhi1998**
- Remove unused log formatter
- Add more logging statement where needed
- Change log levels where appropriate

### CODE REVIEW

File: include/monad/state2/state.hpp [hunk 1]
```diff
@@ -381,6 +389,8 @@ struct State
     // EVMC Host Interface
     [[nodiscard]] bytes32_t get_block_hash(int64_t const number) const noexcept
     {
+        LOG_DEBUG("get_code_size: {}", number);
```
Comment (zander-xyz) [inline include/monad/state2/state.hpp]: this log is wrong

File: include/monad/state2/state.hpp [hunk 2]
```diff
@@ -201,7 +205,7 @@ struct State
 
     void destruct_suicides() noexcept
     {
-        LOG_DEBUG("destruct_suicides");
+        LOG_DEBUG("{}", "destruct_suicides");
```
Comment (zander-xyz) [inline include/monad/state2/state.hpp]: why this change?

File: include/monad/state2/state.hpp (hunk 1 above)
Comment (tongzhi1998) [reply include/monad/state2/state.hpp]: It is indeed. Thanks for catching that!

File: include/monad/state2/state.hpp (hunk 2 above)
Comment (tongzhi1998) [reply include/monad/state2/state.hpp]: I was intending to match it with https://github.com/monad-crypto/monad/blob/main/include/monad/state2/state.hpp#L213, but I realized that I should change the other one inside
